xmodule/library_content_block.py

```diff
@@ -485,6 +519,45 @@ def get_user_id(self):
             user_id = None
         return user_id
 
+    def render_children(self, context, fragment, can_reorder=False, can_add=False):
+        """
+        Renders the children of the module with HTML appropriate for Studio. If can_reorder is True,
+        then the children will be rendered to support drag and drop.
+        """
+        contents = []
+        for child in self.get_children():  # pylint: disable=no-member
+            if can_reorder:
+                context['reorderable_items'].add(child.location)
+            context['can_add'] = can_add
+            context['is_selected'] = str(child.location) in self.candidates
+            rendered_child = child.render(StudioEditableBlock.get_preview_view_name(child), context)
+            fragment.add_fragment_resources(rendered_child)
+
+            contents.append({
+                'id': str(child.location),
+                'content': rendered_child.content
+            })
+
+        fragment.add_content(self.runtime.service(self, 'mako').render_template("studio_render_children_view.html", {  # pylint: disable=no-member
+            'items': contents,
+            'xblock_context': context,
+            'can_add': can_add,
+            'can_reorder': can_reorder,
+        }))
+
+    def get_child_blocks(self):
+        """
+        Return only the subset of our children relevant to the current student.
+        """
+        return list(self._get_selected_child_blocks())
+
+    @XBlock.handler
+    def get_block_ids(self, request, suffix=''):  # lint-amnesty, pylint: disable=unused-argument
+        """
+        Return source_block_ids for selection.
+        """
+        return Response(json.dumps({'source_block_ids': self.candidates}))
```

Comment:
get_block_ids, selection, candidates...

please be consistent with terminology. These are not just any old block ids, and they are not the selection either. They are the candidates.